packages/core/src/components/forms/_controls.scss

```diff
@@ -674,4 +674,129 @@ $control-checked-background-color-active: nth(map-get($button-intents, "primary"
+  function as a purely controlled component, because it needs to adjust its own copy of the `value` internally
+  in response to keyboard and mouse interactions.
+
+  ##### Uncontrolled mode
+
+  To use the numeric stepper in uncontrolled mode, simply pass an `onUpdate` callback into the component
+  to extract the value as the component changes it in response to keyboard and/or mouse interactions. The
+  component will invoke the callback after every value change, giving you the new value as a `string`.
+
+  ```
+  const { NumericStepper } = BlueprintComponents;
+
+  export class NumericStepperExample extends React.Component<{}, {}> {
+      public render() {
+          return (
+              <NumericStepper onUpdate={this.handleUpdate} />
+          );
+      }
+
+      private handleUpdate = (value: string) => {
+          console.log(value);
+      }
+  }
+  ```
+
+  ##### Semi-controlled mode
+
+  Althrough the numeric stepper will always need to maintain state to preoprly adjust its value based on user
+  interactions, you can overwrite the current value by passing in a new `value` as a prop. Together with the
+  `onUpdate` and `onDone` callbacks, this makes it possible to extend the component's basic functionality in
+  a semi-controlled fashion. For instance, we could add support for evaluating mathematical expressions as
+  follows by taking in the expression via an `onDone` callback, evaluating it with some third-party library,
+  then passing the result back into the numeric stepper:
+
+  <div class="pt-callout pt-intent-primary pt-icon-info-sign">
+    By default, non-numeric values will be reset to `0` when the stepper loses focus or when the user
+    presses `Enter`. However, if you provide an `onDone` callback, the stepper will pass the current value
+    through to that callback unaffected, side-stepping the default validation behavior. (Otherwise,
+    mathematical expressions, which may include non-numeric characters, would be reset to `0` before those
+    expressions could be evaluated.)
+  </div>
+
+  ```
+  const { NumericStepper } = BlueprintComponents;
+
+  interface IState {
+      value?: number | string;
+  }
+
+  export class NumericStepperExample extends React.Component<{}, IState> {
+
+      public constructor() {
+          super();
+          this.state = { value: null };
+      }
+
+      public render() {
+          return (
+              <NumericStepper
+                  onDone={this.handleDone}
+                  value={this.state.value}
+              />
+          );
+      }
+
+      private handleDone = (value: string) {
+          if (SomeLibrary.isNumber(value)) {
+              return;
+          }
+          const result = SomeLibrary.evaluateMathExpression(value);
+          this.setState({ value: result });
+      }
+  }
+  ```
+
+  @interface INumericStepperProps
+
+  Weight: -10
+
+  Styleguide components.forms.numeric-stepper.js
+  */
```

Comment:
yeah, that was the right thing to do in this PR @cmslewis. further refactoring should be done separately